### PR TITLE
Added named Data views

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Adds `name: string;` to a Data view.
Adds the 'views' property to Data:`<Data views={[]} />`, which takes an array of views with names.
Adds a DataView component to allow the caller to select one of the named views.
Changes Data to include DataView in the Toolbar.
Changes DataForm to include the selected view name using the '_view' key in the form value object.

#### Where should the reviewer start?

Data/index.d.ts

#### What testing has been done on this PR?

Added unit tests
Added stories

#### How should this be manually tested?

stories

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [x] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?

Ongoing enhancement to `<Data />`, next step is the ability to allow the user to create new named views.

#### Do the grommet docs need to be updated?

yes

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/Data/stories/Simple.js [hunk 1]
```diff
@@ -19,7 +19,20 @@ export const Simple = () => (
       status="info"
       message="Data is in 'beta'. The API surface is subject to change."
     />
-    <Data data={DATA} toolbar>
+    <Data
```
Comment (jcfilben) [inline src/js/components/Data/stories/Simple.js:22]: If I select a view and then attempt to make changes to the filters, my changes are not applied.

Also searching doesn't work while I have a view selected.

https://user-images.githubusercontent.com/54560994/209198305-9adb9048-6f08-46f4-ad14-bb6dc4f26386.mp4

File: src/js/components/DataView/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+## DataFilter
+Documentation for this component: https://v2.grommet.io/datafilter
```
Comment (jcfilben) [inline src/js/components/DataView/README.md:2]:
```suggestion
## DataView
Documentation for this component: https://v2.grommet.io/dataview
```

File: src/js/components/DataView/stories/Simple.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  // <Grommet theme={...}>
+  <Box fill align="center" justify="start" pad="large">
+    <Data
+      data={DATA}
+      views={[
+        { name: 'latest', sort: { property: 'date', direction: 'desc' } },
+        { name: 'behind', properties: { percent: { min: 0, max: 30 } } },
+      ]}
+    >
+      <DataView />
```
Comment (jcfilben) [inline src/js/components/DataView/stories/Simple.js:19]: If DataView is used on its own (not within toolbar) I'm wondering if we should provide a way to clear the selected view.

File: src/js/components/Data/stories/Simple.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/Data/stories/Simple.js:22]: Thanks for catching. These should be fixed now.

File: src/js/components/DataView/stories/Simple.js (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/DataView/stories/Simple.js:19]: Would the "Clear filters" control be enough?

Comment (jcfilben) [reply src/js/components/DataView/stories/Simple.js:19]: Yes

File: src/js/components/Data/stories/Simple.js (hunk 1 above)
Comment (jcfilben) [inline src/js/components/Data/stories/Simple.js:22]: https://user-images.githubusercontent.com/54560994/210440333-92af151d-0781-43ef-b1aa-de7671c640d3.mp4

After selecting a view and then searching, the filters from the view are still applied but the view is no longer shown as active in the Select. This makes it a little confusing what filters are applied and how to get back to where you started

Comment (ericsoderberghp) [reply src/js/components/Data/stories/Simple.js:22]: Good feedback. This should be addressed now.

**Review by MikeKingdom — APPROVED** [review]
Looks good!

File: src/js/components/Data/stories/Simple.js (hunk 1 above)
Comment (jcfilben) [reply src/js/components/Data/stories/Simple.js:22]: I'm no longer seeing this issue with the 'latest' view, but I am still seeing the issue on the 'behind' view

Comment (ericsoderberghp) [reply src/js/components/Data/stories/Simple.js]: Good catch. Should be fixed now.

**Review by jcfilben — COMMENTED** [review]
In the Data/Views story if I apply a filter to the percent or paid field, it correctly applies it but I don't get a 'clear filters' button or badge indication the number of filters applied. If I apply a filter to location or date everything works as expected.

https://user-images.githubusercontent.com/54560994/213002689-c5fc27a4-47cb-43c4-afeb-a67512741ca2.mp4

**Comment by ericsoderberghp** [comment]
> In the Data/Views story if I apply a filter to the percent or paid field, it correctly applies it but I don't get a 'clear filters' button or badge indication the number of filters applied. If I apply a filter to location or date everything works as expected.
> 
>  Jan-17-2023.13-13-59.mp4

That appears to be a pre-existing issue, as the Simple story has the same problem.

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**